bin/cucc.in

```diff
@@ -0,0 +1,287 @@
+
+    # nvcc allows -std=c++ mixed with C files which in case the option is
+    # ineffective for C files.
+    parser.add_argument(
+        '-std', '--std', choices=['c++03', 'c++11', 'c++14', 'c++17', 'c++12'],
+        help="Specify language standard for CUDA and C++ inputs.")
+
+    # The nvcc differs from clang, hipcc and other compilers by that the
+    # option applies globally and only the last option is considered.
+    parser.add_argument('-x', '--x', choices=['c', 'c++', 'cu'],
+                        help="Specify language of the input files")
+
+    parser.add_argument("-default-stream", "--default-stream",
+                        choices=['legacy', 'null', 'per-thread'])
+
+    parser.add_argument("--version", action='store_true')
+    parser.add_argument("-v", "--verbose", action='store_true')
+
+    # TODO: define __CUDA_ARCH__ and __CUDA_ARCH_LIST__ based on
+    #       '-arch' values.
+    parser.add_argument("-arch", action=IgnoredOption)
+    # TODO: handle -fmad. Investigate what is the nvcc's default for this too.
+    parser.add_argument('-fmad', '--fmad', choices=['true', 'false'],
+                        action=IgnoredOption)
+    parser.add_argument("-maxrregcount", "--maxrregcount",
+                        action=IgnoredOption, help="Ignored option.")
+    parser.add_argument("--keep", nargs=0, action=IgnoredOption)
+    parser.add_argument("--keep-dir", action=IgnoredOption)
+    parser.add_argument("--generate-code", action=IgnoredOption)
+    parser.add_argument("-G", "--device-debug", nargs=0, action=IgnoredOption,
+                        help="Generate debug information for device code.")
+    # "The compiler has an option (-use_fast_math) that forces each
+    # function in Table 9 to compile to its intrinsic counterpart. "
+    # [CUDA PM v12.2 13.2].
+    #
+    # CUDA intrinsics have defined numerical accuracy. This means -ffast-math
+    # is potentially more relaxed than --use_fast_math so we may not use it.
+    parser.add_argument("-use_fast_math", "--use_fast_math",
+                        nargs=0, action=IgnoredOption)
+
+    return parser
+
+def get_hip_path() -> str:
+    """Get HIP path
+    """
+    return "@CUCC_HIP_PATH@"
+
+def get_hipcc() -> str:
+    """Get path to hipcc executable
+    """
+    return get_hip_path() + "/bin/hipcc"
+
+def get_cuda_include_dir() -> str:
+    """Return include directory for chipStar's CUDA headers.
+    """
+    return get_hip_path() + "/include/cuspv"
+
+def get_cuda_library_dir() -> str:
+    """Return directory for chipStar's CUDA libraries.
+    """
+    return get_hip_path() + "/lib"
+
+def determine_input_languages(arg_list : list[str], xmode=None) -> set[str]:
+    """Determine input language modes from the argument list
+
+    The 'xmode' specifies value of the last '-x' option if it was specified.
+    returns a set of languages (c, c++ and cu) involved.
+    """
+
+    # '-x' applies globally - unlike in other compilers usually.
```

Comment:
This is an nvcc specific nuance, correct?

Reply:
Yes.